### PR TITLE
GitHub workflows server client

### PR DESCRIPTION
**Pull request by jonas27**

### CODE REVIEW

File: .github/workflows/client.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: Go Client Tests
+
+on:
+  push:
+    branches: ["*"]
+  pull_request:
+    branches: [$default-branch]
```
Comment (avorima) [inline .github/workflows/client.yaml:7]: I can't find this in the docs, have you used this before?

Comment (jonas27) [reply .github/workflows/client.yaml:7]: see https://github.com/actions/starter-workflows/blob/main/ci/go.yml

Comment (avorima) [reply .github/workflows/client.yaml:7]: No wait the actions are reacting on the push action not pull_request. Change branches to the `$default-branch`

Comment (jonas27) [reply .github/workflows/client.yaml:7]: I wanted to have tests run before opening a PR. Obviously that makes tests on PR redundant, but I was not sure if they show up inside the PR as well.

**Pull request closed without merging**